### ISSUE TITLE
JOIN broken after last merge.

### ISSUE THREAD

**Issue by MattiSony** [night/tmi-irc-relay#9]
After the last merge branch it seems like something is wrong with joining channels, when you JOIN a channel it just continually joins it like it's looping. I get this error in the console when it happens "{ [Error: This socket has been ended by the other party] code: 'EPIPE' }" - Which i'm guessing is Twitch disconnecting me because it's sending to many JOINs. I've tried it with mIRC and Hexchat.

**Comment by 3ventic** [comment]
Have you tried with less channels? Twitch will disconnect you if you try to join too many channels too quickly, not only for the 50/10s limitation, but for the send buffer filling up with too much info to send to you as well.

**Comment by MattiSony** [comment]
I tried to join 1 channel. The Relay loops it until Twitch disconnects me.
```
-> tmi.twitch.tv JOIN #mattisony
<- :mattisony!mattisony@mattisony.tmi.twitch.tv JOIN #mattisony
-> tmi.twitch.tv MODE #mattisony
<- :mattisony!mattisony@mattisony.tmi.twitch.tv JOIN #mattisony
<- :mattisony.tmi.twitch.tv 353 mattisony = #mattisony :mattisony
<- :mattisony.tmi.twitch.tv 366 mattisony #mattisony :End of /NAMES list
<- :Twitch MODE #mattisony +~ov mattisony mattisony mattisony
-> tmi.twitch.tv MODE #mattisony
<- @broadcaster-lang=en;r9k=0;slow=0;subs-only=0 :tmi.twitch.tv ROOMSTATE #mattisony
<- :mattisony!mattisony@mattisony.tmi.twitch.tv JOIN #mattisony
<- @broadcaster-lang=en;r9k=0;slow=0;subs-only=0 :tmi.twitch.tv ROOMSTATE #mattisony
<- :mattisony!mattisony@mattisony.tmi.twitch.tv JOIN #mattisony
<- @broadcaster-lang=en;r9k=0;slow=0;subs-only=0 :tmi.twitch.tv ROOMSTATE #mattisony
<- :mattisony!mattisony@mattisony.tmi.twitch.tv JOIN #mattisony
<- @broadcaster-lang=en;r9k=0;slow=0;subs-only=0 :tmi.twitch.tv ROOMSTATE #mattisony
<- :mattisony!mattisony@mattisony.tmi.twitch.tv JOIN #mattisony
<- @broadcaster-lang=en;r9k=0;slow=0;subs-only=0 :tmi.twitch.tv ROOMSTATE #mattisony
```
There's about 40 more lines with that^

**Comment by Crosant** [comment]
Can confirm, this happens everytime

**Comment by night** [comment]
Fixed by #10
